pkg/api/controller.go

```diff
@@ -3511,6 +3511,10 @@ func (c *Controller) GetSetupState(w http.ResponseWriter, r *http.Request) {
 		writeError(w, r, http.StatusInternalServerError, err)
 		return
 	}
+
+	if !c.Config.IsAuthTypeAPI() && savedState == auth.SetupStateNotInitialized {
```

Comment:
Suggest to merge the two condition into if/else
```go
state := string(savedState)
// no need to create an admin user if users are managed externally
if c.Config.IsAuthTypeAPI() {
    state = string(auth.SetupStateInitialized)
} else if savedState == auth.SetupStateNotInitialized {
    c.Collector.CollectEvent(stats.Event{Class: "global", Name: "preinit", Client: httputil.GetRequestLakeFSClient(r)})
}
```

Reply:
Done. Thanks!